static/src/javascripts/projects/commercial/modules/user-features.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+import { getCookie, removeCookie, addCookie } from 'lib/cookies';
+import config from 'lib/config';
+import fetchJson from 'lib/fetch-json';
+import identity from 'common/modules/identity/api';
+// Persistence keys
+const USER_FEATURES_EXPIRY_COOKIE = 'gu_user_features_expiry';
+const PAYING_MEMBER_COOKIE = 'gu_paying_member';
+const AD_FREE_USER_COOKIE = 'GU_AFU';
+
+const userHasData = (): ?string =>
+    getCookie(USER_FEATURES_EXPIRY_COOKIE) ||
+    getCookie(PAYING_MEMBER_COOKIE) ||
+    getCookie(AD_FREE_USER_COOKIE);
+
+const persistResponse = (JsonResponse: () => void) => {
+    const expiryDate = new Date();
+    expiryDate.setDate(expiryDate.getDate() + 1);
+    addCookie(USER_FEATURES_EXPIRY_COOKIE, expiryDate.getTime().toString());
+    addCookie(PAYING_MEMBER_COOKIE, !JsonResponse.adblockMessage);
+    addCookie(AD_FREE_USER_COOKIE, JsonResponse.adFree);
+};
+
+const deleteOldData = (): void => {
+    // We expect adfree cookies to be cleaned up by the logout process, but what if the user's login simply times out?
+    removeCookie(USER_FEATURES_EXPIRY_COOKIE);
+    removeCookie(PAYING_MEMBER_COOKIE);
+    removeCookie(AD_FREE_USER_COOKIE);
+};
+
+const requestNewData = (): void => {
+    fetchJson(`${config.page.userAttributesApiUrl}/me/features`, {
+        mode: 'cors',
+        credentials: 'include',
+    })
+        .then(persistResponse)
+        .catch(() => {});
+};
+
+const featuresDataIsMissing = (): boolean =>
+    !getCookie(USER_FEATURES_EXPIRY_COOKIE) || !getCookie(AD_FREE_USER_COOKIE);
+
+const featuresDataIsOld = (): boolean => {
+    const featuresExpiryCookie = getCookie(USER_FEATURES_EXPIRY_COOKIE);
+    const featuresExpiryTime = parseInt(featuresExpiryCookie, 10);
+    const timeNow = new Date().getTime();
+    return timeNow >= featuresExpiryTime;
+};
+
+const userNeedsNewFeatureData = (): boolean =>
+    featuresDataIsMissing() || featuresDataIsOld();
+
+const userHasDataAfterSignout = (): boolean =>
+    !identity.isUserLoggedIn() && !!userHasData();
```

Comment:
Nit: 
The naming of `userHasData()` implies a boolean return, but the actual return will either be a `string` or `null`.
This seems to be the only usage of `userHasData()`? ...Could we refactor the function to return a boolean (line 11), rather than casting the result to a boolean at this point?

That way we can be more explicit with the return type, and avoid passing Schrodinger's variable (is it truthy, or falsy?) between functions:

```
const userHasData = (): boolean =>
   !!getCookie(USER_FEATURES_EXPIRY_COOKIE) ||
   !!getCookie(PAYING_MEMBER_COOKIE) ||
   !!getCookie(AD_FREE_USER_COOKIE);
```

However, YMMV, 😅 and this may be a better target for a subsequent PR?